### DISCUSSION TITLE
Update on status of iOS companion app

### DISCUSSION THREAD

**Discussion by mprice43** [mKeRix/room-assistant#796] [Q&A]
Hi @mKeRix ,
Thanks for your hard work on room-assistant. It's a really cool app and I have it working as a HA addon, Pi 3b, and Pi Zero.  I use motion sensors to detect room entry and RA to check for presence before allowing the light to time out. I first tried configuring the Classic Bluetooth integration and found it slow and a bit hit and miss. I switched to BLE and run Beacon Maker on my Galaxy watch and Beacon Simulator on my partner's android phone and now it's pretty rock solid. Now I need to tackle my daughter's iPhone so I am trying to gather information on the iOS companion App for RA.
Testflight usually requires an invitation to get the app. Can I get an invitation please?
Also I believe Testflight apps are limited to 90days. Is there a workaround for this?
I notice that the instanceBeacon settings are part of v2.18.4. Does this mean that the beta version is no longer required to run the iOS companion app? 
Keep up the great work. You're a legend.

**Comment by CobraDunn** [comment]
@mprice43 - The companion app has been officially released on the app store. Search for room-assaistant.

**Comment by mKeRix** [comment]
As mentioned already, the app is available in the App Store since a little while now - you can also find a direct link to it in the [docs](https://www.room-assistant.io/integrations/bluetooth-low-energy.html#tracking-ios-devices).

The instance beacon feature is used to support the auto toggling feature of the companion app, but that is turned off by default. Note that this will be dropped/replaced by something similar in upcoming versions, since it caused issues for many users (#763).

**Reply by ivanvach** [reply]
Yes, I'm using the released version of room-assistant companion app for iPhone. Unfortunately  I can't make it working with my room-assistant running on RPi. Here is a fraction of the log

7/24/2021, 00:13:03 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:13:06 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:13:16 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:13:16 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:13:43 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:13:46 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:13:56 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:13:56 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:14:23 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:14:23 - info - BluetoothLowEnergyService: Discovered new BLE peripheral 60a263c880a6 with ID 60a263c880a6 and RSSI -87
7/24/2021, 00:14:24 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:14:34 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:14:34 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:15:03 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:15:04 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:15:14 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:15:14 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:15:43 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:15:44 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:15:54 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:15:54 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:16:23 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:16:26 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:16:36 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:16:36 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:17:03 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:17:03 - info - BluetoothLowEnergyService: Discovered new BLE peripheral 12c874f0028a with ID 12c874f0028a and RSSI -87
7/24/2021, 00:17:04 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:17:14 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:17:14 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:17:43 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:17:44 - info - BluetoothLowEnergyService: Discovered new BLE peripheral 5b7d44b29423 with ID 5b7d44b29423 and RSSI -55
7/24/2021, 00:17:44 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:17:54 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:17:54 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:18:23 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:18:24 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:18:34 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:18:34 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:19:03 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:19:04 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:19:14 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:19:14 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:19:43 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:19:46 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:19:56 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:19:56 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:20:23 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:20:24 - info - BluetoothLowEnergyService: Discovered new BLE peripheral 58cc4662e17f with ID 58cc4662e17f and RSSI -99
7/24/2021, 00:20:25 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:20:35 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:20:35 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out
7/24/2021, 00:21:03 - info - BluetoothService: Detected unusually long lock on Bluetooth adapter 0, force unlocking
7/24/2021, 00:21:07 - info - BluetoothLowEnergyService: Attempting app discovery for tag 49b9a1b9c135
7/24/2021, 00:21:17 - error - BluetoothService: Failed to connect to 49:b9:a1:b9:c1:35: timed out
7/24/2021, 00:21:17 - warn - BluetoothLowEnergyService: Failed to discover companion app ID due to error: timed out

**Reply by ivanvach** [reply]
Please help as I'm very enthusiastic about this app.

**Reply by mKeRix** [reply]
Are you using an external (USB) Bluetooth adapter or the integrated one of the Pi? Looks like something is off with the adapter, any connections seem to fail and make the adapter stall.

**Reply by ivanvach** [reply]
RPi is 2-nd generation. I do use the internal adapter. I've made progress actually and it works intermittently now. Starts and stops working for no apparent reason. iPhone is iPhone11 Pro running iOS 14.6 
I do have RPi 3-d gen. Pls advise if switching to RPi 3-rd gen will improve or using external BLE adapter will improve things.

**Comment by neofraz03** [comment]
So is the iOS Room-Assistant app dead?  I have it on my phone since early 2024, but now I'm trying to track my wife's iPhone and I cannot locate the app on the AppStore.
